generator/src/generator/namespace.rs

```diff
@@ -3577,7 +3591,11 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                         outln!(out, "}}");
                     }
                     outln!(out, "match parse_result {{");
-                    outln!(out.indent(), "None => Err(ParseError::InvalidValue),");
+                    outln!(
+                        out.indent(),
+                        "None => Ok(({}::InvalidValue(switch_expr), outer_remaining)),",
```

Comment:
Instead of returning `outer_remaining`, it might be better to return `&outer_remaining[..0]`, i.e. an empty slice. That shouldn't make much of a difference currently, but it seems more future-proof to me: If anything tries to put a field behind a `<switch>`, parsing will fail in this case (no data left to parse) instead of silently trying to parse the data of the switch.